bin/desi_tile_redshifts

```diff
@@ -0,0 +1,343 @@
+        keep = (t['OBSTYPE'] == 'science') & (t['TILEID'] >= 0)
+        if 'LASTSTEP' in t.colnames:        
+            keep &= (t['LASTSTEP'] == 'all')
+
+        t = t[keep]
+        exptables.append(t['TILEID', 'NIGHT', 'EXPID'])
+
+    return vstack(exptables)
+
+#-------------------------------------------------------------------------
+log = get_logger()
+
+#- If --tileid, --night, and --expid are all given, create exptable
+if ((args.tileid is not None) and
+    (args.night is not None) and (len(args.night) == 1) and
+    (args.expid is not None)):
+    log.info('Creating exposure table from --tileid --night --expid options')
+    exptable = Table()
+    exptable['EXPID'] = args.expid
+    exptable['NIGHT'] = args.night[0]
+    exptable['TILEID'] = args.tileid
+
+    if args.explist is not None:
+        log.warning('Ignoring --explist, using --tileid --night --expid')
+
+#- otherwise load exposure tables for those nights
+elif args.explist is None:
+    if args.night is not None:
+        log.info(f'Loading production exposure tables for nights {args.night}')
+    else:
+        log.info(f'Loading production exposure tables for all nights')
+
+    exptable = _read_minimal_exptables(args.night)
+
+else:
+    log.info(f'Loading exposure list from {args.explist}')
+    if args.explist.endswith('.fits'):
+        exptable = Table.read(args.explist, format='fits')
+    elif args.explist.endswith('.csv'):
+        exptable = Table.read(args.explist, format='ascii.csv')
+    elif args.explist.endswith('.ecsv'):
+        exptable = Table.read(args.explist, format='ascii.ecsv')
+    else:
+        exptable = Table.read(args.explist, format='ascii')
+
+    if args.night is not None:
+        keep = np.in1d(exptable['NIGHT'], args.night)
+        exptable = exptable[keep]
+
+#- Filter exposure tables by exposure IDs or by tileid
+#- Note: If exptable was created from --expid --night --tileid these should
+#- have no effect, but are left in for code flow simplicity
+if args.expid is not None:
+    keep = np.in1d(exptable['EXPID'], args.expid)
+    exptable = exptable[keep]
+    expids = np.array(exptable['EXPID'])
+    tileids = np.unique(np.array(exptable['TILEID']))
+
+    #- if provided, tileid should be redundant with the tiles in those exps
+    if args.tileid is not None:
+        if not np.all(exptable['TILEID'] == args.tileid):
+            log.critical(f'Exposure TILEIDs={tileids} != --tileid={args.tileid}')
+            sys.exit(1)
+ 
+elif args.tileid is not None:
+    keep = (exptable['TILEID'] == args.tileid)
+    exptable = exptable[keep]
+    expids = np.array(exptable['EXPID'])
+    tileids = np.array([args.tileid,])
+
+else:
+    tileids = np.unique(np.array(exptable['TILEID']))
+
+#- anything left?
+if len(exptable) == 0:
+    log.critical(f'No exposures left after filtering by tileid/night/expid')
+    sys.exit(1)
+
+#- default group is the YEARMMDD night, but must be given if combining nights
+if args.group is None:
```

Comment:
again if we want to support single exposures this could auto-detect a single exposure and name group={expid}. Or we could require the user to define it on the command line. Up to you, but wanted to mention it